### ISSUE TITLE
Давайте придумаем как лучше сделать кнопку апвоута в конце поста

### ISSUE THREAD

**Issue by vas3k** [vas3k/vas3k.club#315]
Мы все вроде согласились, что в конце поста тоже нужна кнопка +1. Вопрос как ее сделать.

Плавающая или фикс? Как отображать чтобы не раздражала? Будет ли разница на мобиле и десктопе?

Короче, нужны идеи или рефы. На каких сайтах вы видели норм реализацию и хотели бы так же? Пойдут любые идеи, брейнштормим.

**Comment by FMajesty** [comment]
Как по мне, классно было бы запилить плавающую кнопочку.
На мобилках где-нибудь снизу справа сделать. А вот на десктопе хз даже. На том же месте, где сейчас кнопка, только сделать так, чтобы плыла вместе с экраном?👀

**Comment by nlopin** [comment]
Давайте сделаем кнопку в конце поста. Это просто и быстро.

Минусы плавающей:
* непонятно, как показывать на мобиле. На узком экране телефона она будет мешаться
* много неочевидных интерфейсных вопросов: что делать если пост короткий? нужно ли ее останавливать в месте, где кончается пост и начинается комментарии?
* все эти интерфейсные вопросы порождают сложность в реализации, а мы здесь опенсорсим и будем долго делать

Сырой набросок:
<img width="40" alt="image" src="https://user-images.githubusercontent.com/1469636/88232844-e0ec2100-cc76-11ea-9e48-a9323e66973e.png">

**Comment by AshotPilosian** [comment]
Я за фикс кнопку, т.к. в плавающей нет большого смысла, она только будет отвлекать от поста или будет незаметна, как это происходит с кнопкой меню на смартфоне.

Верхнюю кнопку можно оставить, но для меня она большого смысла не несет, потому что я открыл пост, я еще не знаю, что там, ставить лайк я не буду. Я дочитал пост, у меня есть мнение и тут на тебе, кнопка снизу! 

С точки зрения вида, мне нравится, как это сделано на медиуме, например, по-сути, апвоуты — те же клэпы, просто другая иконка, на смартфоне можно слева между постом и инфой об авторе расположить (а справа потом можно кнопку подписаться на комменты, например или шейры в соцсети или еще что то), на десктопе пока не знаю.

**Comment by devcooch** [comment]
Я согласен с Ашотом - существующую кнопку просто убрать вниз. Как вариант можно показывать количество голосов вверху, а кнопку внизу.. Хотя это тоже не особо логично.
Если же делать плавающую, то на мобиле не очень будет, как уже сказали выше.

**Comment by Windrushfarer** [comment]
Мне нравится текущая кнопка, мне кажется она отлично вписывается. Так давайте просто в конце бахнем кнопку по типу «Мне понравилось 👍» и достаточно.
Как конечно крайний вариант можно попробовать сделать фиксированную плашку внизу, которая будет всегда там по аналогии с тем как это на постах в dev.to, вот только сейчас это оверкилл т.к есть только апвоуты. Возможно когда появятся закладки будет актуальнее.

**Comment by skywinder** [comment]
я за кнопку внизу. Кнопка вверху только мешается. 
Пару раз случайно при скролле апвоут ставил, когда еще отменить его было нельзя.
Пример как сделать да хоть эта таска на гитхабе со смайликами реакциями. Хоть фб-вк-медиум.

**Comment by ivannikovdv** [comment]
+кнопки внизу еще в том, что пользователю меньше соблазна сразу ее нажать из-за например количества апвойтов которое он сразу видет (эффект толпы), хотя он это видет еще на экране списка тем, но все же и он успеет текст прочитать до того как оценить или передумать оценивать.

**Comment by ivannikovdv** [comment]
+кнопки внизу еще в том, что пользователю меньше соблазна сразу ее нажать из-за например количества апвойтов которое он сразу видит (эффект толпы), хотя он это видит еще на экране списка тем, но все же и он успеет текст прочитать до того как оценить или передумать оценивать.

**Comment by husky-dev** [comment]
Я за кнопку в низу и за плавающюю кнопку сбоку (но только если размер экрана позволяет). Так например реализовано на Медиуме.

![Nov-20-2020 11-38-42 optimize](https://user-images.githubusercontent.com/1960658/99793958-1bd16900-2b32-11eb-8135-01a35bc8548b.gif)

Для чего:

- Часто сталкиваюсь с проблемой, что пост понравился и хочется лайкнуть. Но приходится скролить на верх.
- Кнопка внизу поста может решить эту проблему. Но слишком большая будет занимать много места на мобайле и как бы провоцировать лайнуть (имхо).
- Порой читаешь пост и сразу понимаешь, что хочешь лайкнуть. Скролишь вверх что бы лайкнуть и не забыть. При этом теряешь место где остановился. Плавающая кнопка решила бы этот вопрос.
- Еще юзкейс: читаешь пост, все нравится, и дочитав до конца сразу бросаешься в коменты (там же ж самый сок!). В итоге пост не лайкнут и есть вероятность что кнопка снизу будет проигнорена (в случае если она будет небольшая).
- Плавающая кнопка может повысить уровень вовлеченности людей, которые хотели бы лайкнуть пост, но не сделали этого из-за UX.
- Больше лайков -> Приятней автору -> Больше мотивации писать новые посты.
- "эффект толпы" упомянутый выше можно нивелировать за счет того, что бы не сразу показывать количество апвоутов. Можно показывать кнопку апвоут, а уже после того как пользователь ее нажал показать количество лайков. Но это можно реализовать в следующей итерации. Для начала чисто плавающая и в футере.
- Что бы плавающая меньше мешала чтению можно уменьшать саму кнопку по ходу скрола и сделать немного прозрачней (возвращать прозрачность при наведении).

Набросла эскиз. На гифке оно не так красиво, но в браузере все плавненько 👍 Опять же, это только для случая когда на экране есть место. Для мобайла кнопка не плавает. Под конец скрола текста поста кнопка плавно исчезает, и тогда пользователь уже может лайкнуть с помощью кнопки в конце поста.

![Nov-20-2020 13-04-05 optimize](https://user-images.githubusercontent.com/1960658/99793984-2855c180-2b32-11eb-8125-69b52a8a2cfc.gif)
